functions/go/set-labels/transformer/labels_test.go

```diff
@@ -0,0 +1,332 @@
+package transformer
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/GoogleContainerTools/kpt-functions-sdk/go/fn"
+)
+
+func TestLabelTransformer_ConfigMap_Service(t *testing.T) {
+	functionConfig := `
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: my-config
+data:
+  app: myApp
+  quotedBoolean: "true"
+  quotedFruit: "peach"
+  unquotedBoolean: true
+  env: production
+`
+	input := `
+apiVersion: v1
+kind: Service
+metadata:
+  name: whatever
+spec:
+  selector:
+    a: b
+`
+
+	expected := `
+apiVersion: v1
+kind: Service
+metadata:
+  name: whatever
+  labels:
+    app: myApp
+    env: production
+    quotedBoolean: "true"
+    quotedFruit: peach
+    unquotedBoolean: "true"
+spec:
+  selector:
+    a: b
+    app: myApp
+    env: production
+    quotedBoolean: "true"
+    quotedFruit: peach
+    unquotedBoolean: "true"
+`
+
+	transformer := LabelTransformer{}
+	config, _ := fn.ParseKubeObject([]byte(functionConfig))
+	_ = transformer.Config(config)
+	result, _ := fn.ParseKubeObject([]byte(input))
+	err := transformer.Transform(fn.KubeObjects{result})
```

Comment:
super nit: you can use `ResourceList` to define the inputs (which contains both functionConfig and the KRM resources) and call `ParseResourceList`

Reply:
you can make the suggested change in a follow-up PR.

Reply:
I am curious why we need to use `ResourceList` here. Is it because it is more intuitive for users, since each operation is considered transformation on resoureList? Because 
`data := fmt.Sprintf("functionConfig: %s\nitems: \n... %s", functionConfig, singleItem)` is actually not easy to format with indentation. If only consider the code simplicity, it would be best to parse config and resource separately

Reply:
I was sharing some pseudo code off the top of my head, agree the `fmt.Sprint` is not a good format.
In the fn sdk, we basically defines the entire `ResourceList` as input. the main reason I vote down on parsing the fnConfig and parsing the input resources separately is that KRM function testing more or less need a lot of code to generate testing resource. It's much less maintainable to handle each type of function resource separately than putting them all in the ResourceList in the form of yaml (string) format. 

Also, this gives users a more directly understanding about what the expected input is (ResourceList Yaml) and what the output is (modified ResourceList).